.github/workflows/docker-build-static.yml

```diff
@@ -151,93 +128,109 @@ jobs:
             .cargo/registry/index
             .cargo/registry/cache
             .cargo/git/db
-            target
-          key: lldap-bin-aarch64-${{ hashFiles('**/Cargo.lock') }}
+            target/
+          key: lldap-bin-${{ matrix.target }}-${{ hashFiles('**/Cargo.lock') }}
           restore-keys: |
-           lldap-bin-aarch64-
-#      - name: fetch musl-gcc
-#        run: |
-#             wget -c https://musl.cc/aarch64-linux-musl-cross.tgz
-#             tar zxf ./x86_64-linux-musl-cross.tgz -C /opt
-#             echo "/opt/aarch64-linux-musl-cross:/opt/aarch64-linux-musl-cross/bin" >> $GITHUB_PATH
-      - name: add musl aarch64 target
-        run: rustup target add aarch64-unknown-linux-musl
-      - name: build lldap aarch4
-        run: cargo build --target=aarch64-unknown-linux-musl --release -p lldap -p migration-tool
-      - name: check path
-        run: ls -al target/aarch64-unknown-linux-musl/release/
-      - name: upload aarch64 lldap artifacts
+            lldap-bin-${{ matrix.target }}-
+      # armhf not existed in dev image, so install here. can be removed after update dev image
+      - name: Add armhf architecture
+        run: dpkg --add-architecture armhf
+      - name: Install build deps for armhf
+        run: apt update && apt install -y gcc-arm-linux-gnueabihf g++-arm-linux-gnueabihf libc6-armhf-cross libc6-dev-armhf-cross tar ca-certificates
+      - name: Install ${{ matrix.target }} target
+        run: rustup target add ${{ matrix.target }}
```

Comment:
I think we can remove this rustup?

Reply:
Oh the base image updated, should be removed then.